lib/srv/db/redis/engine.go

```diff
@@ -387,6 +387,10 @@ func (e *Engine) isAWSIAMAuthSupported(ctx context.Context, sessionCtx *common.S
 	dbUser := sessionCtx.DatabaseUser
 	ok, err := e.checkUserIAMAuthIsEnabled(ctx, sessionCtx, dbUser)
 	if err != nil {
+		if sessionCtx.Database.GetType() == types.DatabaseTypeElastiCacheServerless {
+			e.Log.DebugContext(e.Context, "Assuming IAM auth is enabled for user", "user", dbUser, "error", err)
+			return true
+		}
```

Comment:
Based on this condition, I assume the ElasticCache serverless will always use the IAM auth credentials. Given this, should we consider this a hard failure (or at least have it logged as a warning/error) because the database users are not correctly configured (and this will probably cause them to fail when authenticating)?

nit: What about moving this condition to the `checkUserIAMAuthIsEnabled` function? It already has a database-type condition, and we could have it handle this case as well.